app/src/main/java/com/duckduckgo/app/email/ui/EmailProtectionViewModel.kt

```diff
@@ -0,0 +1,132 @@
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import javax.inject.Provider
+
+class EmailProtectionViewModel(
+    private val emailManager: EmailManager,
+    private val emailService: EmailService,
+    private val workManager: WorkManager,
+    private val waitlistWorkRequestBuilder: WaitlistWorkRequestBuilder,
+) : ViewModel() {
+
+    private val viewStateFlow: MutableStateFlow<ViewState> = MutableStateFlow(ViewState(waitlistState = emailManager.waitlistState()))
+    val viewState: StateFlow<ViewState> = viewStateFlow
+
+    private val commandChannel = Channel<Command>(capacity = 1, onBufferOverflow = BufferOverflow.DROP_OLDEST)
+    val commands = commandChannel.receiveAsFlow()
+
+    sealed class Command {
+        data class OpenUrl(val url: String, val openInBrowser: Boolean) : Command()
+        object ShowErrorMessage : Command()
+        object ShowNotificationDialog : Command()
+    }
+
+    data class ViewState(val waitlistState: AppEmailManager.WaitlistState)
+
+    fun haveADuckAddress() {
+        viewModelScope.launch {
+            commandChannel.send(Command.OpenUrl(url = LOGIN_URL, openInBrowser = true))
+        }
+    }
+
+    fun haveAnInviteCode() {
+        viewModelScope.launch {
+            commandChannel.send(Command.OpenUrl(url = "$SIGN_UP_URL${emailManager.getInviteCode()}", openInBrowser = true))
+        }
+    }
+
+    suspend fun joinTheWaitlist() {
+        runCatching {
+            emailService.joinWaitlist()
+        }.onSuccess {
+            val timestamp = it.timestamp
+            val token = it.token
+            if (timestamp != null && !token.isNullOrBlank()) {
+                joinedWaitlist(it.timestamp, it.token)
+            } else {
+                commandChannel.send(Command.ShowErrorMessage)
+            }
+        }.onFailure {
+            commandChannel.send(Command.ShowErrorMessage)
+        }
+    }
+
+    fun readBlogPost() {
+        viewModelScope.launch {
+            commandChannel.send(Command.OpenUrl(url = ADDRESS_BLOG_POST, openInBrowser = false))
+        }
+    }
+
+    fun readPrivacyGuarantees() {
+        viewModelScope.launch {
+            commandChannel.send(Command.OpenUrl(url = PRIVACY_GUARANTEE, openInBrowser = false))
+        }
+    }
+
+    private fun joinedWaitlist(timestamp: Int, token: String) {
+        viewModelScope.launch {
+            emailManager.joinWaitlist(timestamp, token)
+            viewStateFlow.emit(ViewState(emailManager.waitlistState()))
+            commandChannel.send(Command.ShowNotificationDialog)
+        }
+        workManager.enqueue(waitlistWorkRequestBuilder.waitlistRequestWork(withDelay = false))
+    }
+
+    companion object {
```

Comment:
NIT: maybe we could move this URLs into `AppUrl` class?

Reply:
I'm going to leave them here for now because this is a temporary feature so we will come back here at some point.